### PR TITLE
ST-1315: Change search-bar behaviour

### PR DESCRIPTION
**Pull request by ilyawzrd**
Description:

--

QA-test: ST-1317

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.1.0-pr-156-239560c7.zip

### CODE REVIEW

File: client-app/shared/layout/components/search-bar/search-bar.vue [hunk 1]
```diff
@@ -10,16 +10,15 @@
...
         class="flex-grow mr-4 rounded h-10 px-4 font-medium text-sm outline-none disabled:bg-gray-200"
-        @keyup.enter="search"
+        @keyup.enter="onInputChange"
         @keyup.esc="searchDropdownVisible && hideSearchDropdown()"
-        @input="search"
+        @input="onInputChange"
       />
 
-      <VcButton class="uppercase px-4 !h-10" @click="search">
+      <VcButton class="uppercase px-4 !h-10" @click="searchPhrase.length > 0 && onSearchButtonClick">
```
Comment (trueboroda) [inline client-app/shared/layout/components/search-bar/search-bar.vue:21]: move condition logic to method

**Review by trueboroda — CHANGES_REQUESTED** [review]

File: client-app/shared/layout/components/search-bar/search-bar.vue (hunk 1 above)
Comment (ilyawzrd) [reply client-app/shared/layout/components/search-bar/search-bar.vue:21]: We have same type of logic throughout this component. It's just a precondition to call it

File: client-app/shared/layout/components/search-bar/search-bar.vue [hunk 2]
```diff
@@ -154,7 +153,15 @@ const categoriesColumns = computed<Array<Category[]>>(() => {
...
+  if (searchDropdownVisible) {
+    hideSearchDropdown();
+  }
+
+  callSearch();
+}
+
+function onInputChange() {
   const MAX_LENGTH = 30;
   const MIN_LENGTH = cfg.search_min_chars || 0;
```
Comment (artmakarov) [inline client-app/shared/layout/components/search-bar/search-bar.vue:166]: Maybe `cfg.search_min_chars || 3`

File: client-app/shared/layout/components/search-bar/search-bar.vue (hunk 1 above)
Comment (trueboroda) [reply client-app/shared/layout/components/search-bar/search-bar.vue:21]: Better encapsulate all logic inside the method.

**Review by trueboroda — APPROVED** [review]

**Review by artmakarov — APPROVED** [review]

**Pull request merged**